tests/integration_tests/tests/connection.rs

```diff
@@ -52,7 +52,10 @@ async fn connect_returns_err_via_call_after_connected() {
     let res = client.unary_call(Request::new(Input {})).await;
 
     let err = res.unwrap_err();
-    assert_eq!(err.code(), Code::Unavailable);
+    assert_eq!(err.code(), Code::Unknown);
```

Comment:
I took a deeper dive here - `hyper` stopped exposing `is_connect()` since the connection tooling was moved to `hyper_util`. However, `tonic` wraps connection logic anyways, so I stuck in an (internal) wrapper type to identify connection errors and set the correct status.